client/my-sites/checkout/checkout/recent-renewals.jsx

```diff
@@ -34,37 +34,43 @@ function RecentRenewalListItem( { domain, productName, expiryMoment, translate }
 RecentRenewalListItem.propTypes = {
 	domain: PropTypes.string.isRequired,
 	productName: PropTypes.string.isRequired,
-	expiryMoment: PropTypes.object.isRequired,
+	expiry: PropTypes.object.isRequired,
 	translate: PropTypes.func.isRequired,
 };
 
-function getRecentRenewalProductsMatchingIds( products, ids ) {
+function getRecentRenewalProductsMatchingIds( products, ids, moment ) {
 	const oldestMoment = moment().subtract( 30, 'days' );
 	return products.filter( product => {
 		return (
 			ids.includes( product.productId ) &&
 			product.subscriptionStatus === 'active' &&
 			product.productName &&
-			product.expiryMoment &&
-			product.mostRecentRenewMoment &&
-			product.mostRecentRenewMoment.isAfter( oldestMoment )
+			product.expiryDate &&
+			product.mostRecentRenewDate &&
+			moment( product.mostRecentRenewDate ).isAfter( oldestMoment )
 		);
 	} );
 }
 
-export function RecentRenewals( { purchases, siteId, cart, translate } ) {
+export function RecentRenewals( { purchases, siteId, cart } ) {
+	const translate = useTranslate();
+	const moment = useLocalizedMoment();
+
 	const idsInCart = cart.products ? cart.products.map( product => product.product_id ) : [];
-	const recentRenewals = getRecentRenewalProductsMatchingIds( purchases, idsInCart );
+	const recentRenewals = getRecentRenewalProductsMatchingIds( purchases, idsInCart, moment );
 	const productListItems = recentRenewals.map( product => {
 		const domain = product.isDomainRegistration
 			? product.meta || product.domain
 			: product.includedDomain || product.domain;
+
+		const expiry = moment( product.expiryDate, product.expiryDateFormat );
+
 		return (
 			<RecentRenewalListItem
 				key={ product.id }
 				domain={ domain }
 				productName={ product.productName }
-				expiryMoment={ product.expiryMoment }
+				expiry={ expiry && expiry.format( 'LL' ) }
```

Comment:
I think we should clean up those checks - what is the reason to have them in the first place?

Reply:
My concern was that the `moment()` call could return a falsy value depending on the values provided, and thus this would generate an error accessing the property. After looking at the documentation, this doesn't appear to be possible.